[ADD] Quality: Alerts report (#9403)
odoo/documentation: content/applications/inventory_and_mrp/quality/quality_reporting/quality_alerts_report.rst

=====================
Quality alerts report
=====================

In Odoo's :menuselection:`Quality` app, the quality alerts report displays key statistics about
existing alerts. This includes the number of quality alerts in each stage, the number of alerts
assigned to each quality team, the root causes of the issues that led to alert creation, and more.

The report offers a number of options to help visualize the data, including customizable views and
three different graph types. This allows users to extract deeper insights from their quality alert
processes.

Report dashboard
================

To access the quality alerts report dashboard, navigate to :menuselection:`Quality app --> Reporting
--> Quality Alerts`.

There are five main views that can be used to display quality alert data:

> - Graph view

Reviewer: you could add the icon in front of each of these since we have them =)

Reviewer: You can also add a link to jump to the sextion below for each type.

   The view select buttons on the quality alerts report dashboard. From left to right: Graph, Pivot,
   Kanban, List, and Calendar.

Graph view
----------

To display quality alert data in graph view, click the :icon:`fa-area-chart` :guilabel:`(Graph)`
button in the top-right corner of the dashboard.

There are three chart types that can be used to visualize quality alert data:

- Bar chart
- Line chart
- Pie chart

By default, graph view displays data using the bar chart graph type. To select a different graph
type, click the corresponding button in the top-left corner of the dashboard.

.. figure:: quality_alerts_report/graph-select.png
   :align: center
   :alt: The graph select buttons on the quality alerts report dashboard.

   The graph select buttons on the quality alerts report dashboard. From left to right: Bar Chart,
   Line Chart, and Pie Chart.

Bar chart
~~~~~~~~~

To display quality alert data in a bar chart, click the :icon:`fa-bar-chart` :guilabel:`(Bar Chart)`
button in the top-left corner of the dashboard.

The x-axis of the bar chart lists root causes that led to the creation of one or more alerts (ex.

> *workcenter failure*). Since root causes can be added by users, they may differ from one database to

Reviewer: ```suggestion
*Workcenter Failure*). Since root causes can be added by users, they may differ from one database to
```

The x-axis of the line chart lists root causes that led to the creation of one or more alerts (ex.
workcenter failure). Since root causes can be added by users, they may differ from one database to
another. The y-axis of the bar chart shows the number of quality alerts associated with each root
cause, which is represented by a point on the graph above each stage name. Each of these points is
linked by a line.

.. image:: quality_alerts_report/line-chart.png
   :align: center
   :alt: The line chart graph view on the quality alerts report dashboard.

Pie chart
~~~~~~~~~

To display quality alert data in a pie chart, click the :icon:`fa-pie-chart` :guilabel:`(Pie Chart)`
button in the top-left corner of the dashboard.

Each slice of the pie chart represents the quality alerts associated with a specific root cause, in
a specific stage of the alert review process (ex. workcenter failure alerts in the *In Review*
stage). The size of each slice represents the number of alerts assigned to it, as a percentage of
the total number of alerts.

Each slice of the pie appears in a unique color, and a key at the top of the chart denotes the root
cause and review stage represented by each color.

> To remove a slice from the pie chart, click its name in the key at the top of the chart. After doing

Reviewer: Wow - I never noticed this!

so, the remaining slices expand to fill the space vacated by the removed slice. Click the slice name
again to make the slice reappear on the chart.

.. image:: quality_alerts_report/pie-chart.png
   :align: center
   :alt: The pie chart graph view on the quality alerts report dashboard.

Pivot view
----------

To display quality alert data in pivot view, click the :icon:`oi-view-pivot` :guilabel:`(Pivot)`
button in the top-right corner of the dashboard. The pivot view displays data in the form of a
spreadsheet.

> By default, each column is labeled with the name of a quality alert stage (ex. In Review). Since

Reviewer: ```suggestion
By default, each column is labeled with the name of a quality alert stage (ex. *In Review*). Since
```

By default, each row is labeled with the name of a quality alert team.

The cell at the intersection of a column and row displays a number, which signifies the number of
quality alerts in the column's stage which are assigned to the row's quality alert team.

To change the data displayed by the columns or rows, click the :guilabel:`Total` heading above
either. Then, click it again to open a popover menu, from which a new variable can be selected.

.. example::
   Click the :guilabel:`Total` heading above the columns to remove the stage names. Then, click the
   heading again, and select :guilabel:`Root Cause` from the popover menu. Each column is now titled
   with the name of a root cause.

.. figure:: quality_alerts_report/pivot-view.png
   :align: center
   :alt: The default pivot view on the quality alerts report dashboard.

   The default pivot view on the quality alerts report dashboard.

Kanban view
-----------

To display quality alert data in a Kanban view, click the :icon:`oi-view-kanban`
:guilabel:`(Kanban)` button in the top-right corner of the dashboard.

Kanban view displays all existing quality alerts in a Kanban board, with each alert organized into a

> specific stage (ex. In Review). Since kanban stages are customizable, they may differ from one

Reviewer: ```suggestion
specific stage (ex. *In Review*). Since kanban stages are customizable, they may differ from one
```

Reviewer: I'm not sure if that kanban gets a capital K - I don't HTINK so since you're describing it in general, and not on our acutal UI. Not 100% sure though.

database to another.

Quality alerts can be drag-and-dropped from one stage to another. Doing so changes the status of the
quality alert.

Click on a quality alert to open it and see its details.

.. image:: quality_alerts_report/kanban-view.png
   :align: center
   :alt: The Kanban view on the quality alerts report dashboard.

List view
---------

To display all quality alerts in a list view, click the :icon:`oi-view-list` :guilabel:`(List)`
button in the top-right corner of the dashboard.

> List view displays all existing quality alerts in a list. Click on a quality alert to open it and

Reviewer: You can list the default columns if you want, and list the optional items as well. Just an idea to make this chunk a bit bigger.

=====================
Quality alerts report
=====================

> In Odoo's *Quality* app, the quality alerts report displays key statistics about existing alerts.

Reviewer 2: ```suggestion
In Odoo's *Quality* app, the *Quality Alerts* report displays key statistics about existing alerts.
```
suggesting the use of italics here because it is the name of the reporting page that appears on the UI, but since it's not "presently" on the screen at this point, *italics* would be used.

This includes the number of quality alerts in each stage, the number of alerts assigned to each
quality team, the root causes of the issues that led to alert creation, and more.

The report offers a number of options to help visualize the data, including customizable views and
three different graph types. This allows users to extract deeper insights from their quality alert
processes.

Report dashboard
================

To access the quality alerts report dashboard, navigate to :menuselection:`Quality app --> Reporting
--> Quality Alerts`.

There are five main views that can be used to display quality alert data:

- :ref:`Graph view <quality_alerts_report/graph-view>`
- :ref:`Pivot view <quality_alerts_report/pivot-view>`
- :ref:`Kanban view <quality_alerts_report/kanban-view>`
- :ref:`List view <quality_alerts_report/list-view>`

> - :ref:`Calendar view <quality_alerts_report/calendar-view>`

Reviewer 2: You may want to consider adding in the appropriate icons beside each of these, so the reader can know what the view icons represent. just a thought.

By default, the dashboard displays data in the graph view, using the bar chart graph type. To select
a different view, click on the corresponding button in the top-right corner of the dashboard.

.. figure:: quality_alerts_report/view-select.png
   :align: center
   :alt: The view select buttons on the quality alerts report dashboard.

>    The view select buttons on the quality alerts report dashboard. From left to right: Graph, Pivot,

Reviewer 2: ```suggestion
   The view select buttons on the Quality Alerts report dashboard. From left to right: Graph, Pivot,
```

   Kanban, List, and Calendar.

.. _quality_alerts_report/graph-view:

Graph view
----------

To display quality alert data in graph view, click the :icon:`fa-area-chart` :guilabel:`(Graph)`
button in the top-right corner of the dashboard.

There are three chart types that can be used to visualize quality alert data:

- :ref:`Bar chart <quality_alerts_report/bar-chart>`
- :ref:`Line chart <quality_alerts_report/line-chart>`

> - :ref:`Pie chart <quality_alerts_report/pie-chart>`

Reviewer 2: would you want to consider including the appropriate icons for these options?

> By default, graph view displays data using the bar chart graph type. To select a different graph

Reviewer 2: ```suggestion
By default, the graph view displays data using the bar chart graph type. To select a different graph
```

type, click the corresponding button in the top-left corner of the dashboard.

.. figure:: quality_alerts_report/graph-select.png
   :align: center
   :alt: The graph select buttons on the quality alerts report dashboard.

>    The graph select buttons on the quality alerts report dashboard. From left to right: Bar Chart,

Reviewer 2: ```suggestion
   The graph select buttons on the Quality Alerts report dashboard. From left to right: Bar Chart,
```

   Line Chart, and Pie Chart.

.. _quality_alerts_report/bar-chart:

Bar chart
~~~~~~~~~

To display quality alert data in a bar chart, click the :icon:`fa-bar-chart` :guilabel:`(Bar Chart)`

> button in the top-left corner of the dashboard.

Reviewer 2: ```suggestion
button in the top-left corner of the dashboard, while in graph view.
```

> The x-axis of the bar chart lists root causes that led to the creation of one or more alerts (ex.

Reviewer 2: ```suggestion
The x-axis of the bar chart lists a specific :guilabel:`Root Cause` that led to the creation of one or more alerts (e.g.
```
It's my understanding that we're going with "e.g." in these parenthetical instances...
Also, since Root Cause appears on the screen, I believe a GUI should be applied.

*Workcenter Failure*). Since root causes can be added by users, they may differ from one database to
another. The y-axis of the bar chart shows the number of quality alerts associated with each root
cause, which is represented by a bar above each stage name.

.. image:: quality_alerts_report/bar-chart.png
   :align: center
   :alt: The bar chart graph view on the quality alerts report dashboard.

.. _quality_alerts_report/line-chart:

Line chart
~~~~~~~~~~

To display quality alert data in a line chart, click the :icon:`fa-line-chart` :guilabel:`(Line

> Chart)` button in the top-left corner of the dashboard.

Reviewer 2: ```suggestion
Chart)` button in the top-left corner of the dashboard, while in graph view.
```

> The x-axis of the line chart lists root causes that led to the creation of one or more alerts (ex.

Reviewer 2: ```suggestion
The x-axis of the line chart lists a specific :guilabel:`Root Cause` that led to the creation of one or more alerts (e.g.
```

> Workcenter Failure). Since root causes can be added by users, they may differ from one database to

Reviewer 2: ```suggestion
*Workcenter Failure*). Since root causes can be added by users, they may differ from one database to
```

another. The y-axis of the bar chart shows the number of quality alerts associated with each root
cause, which is represented by a point on the graph above each stage name. Each of these points is
linked by a line.

.. image:: quality_alerts_report/line-chart.png
   :align: center
   :alt: The line chart graph view on the quality alerts report dashboard.

.. _quality_alerts_report/pie-chart:

Pie chart
~~~~~~~~~

To display quality alert data in a pie chart, click the :icon:`fa-pie-chart` :guilabel:`(Pie Chart)`

> button in the top-left corner of the dashboard.

Reviewer 2: ```suggestion
button in the top-left corner of the dashboard, while in graph view.
```

Each slice of the pie chart represents the quality alerts associated with a specific root cause, in

> a specific stage of the alert review process (ex. Workcenter Failure alerts in the *In Review*

Reviewer 2: ```suggestion
a specific stage of the alert review process (e.g. *Workcenter Failure* alerts in the *In Review*
```

stage). The size of each slice represents the number of alerts assigned to it, as a percentage of
the total number of alerts.

Each slice of the pie appears in a unique color, and a key at the top of the chart denotes the root
cause and review stage represented by each color.

To remove a slice from the pie chart, click its name in the key at the top of the chart. After doing
so, the remaining slices expand to fill the space vacated by the removed slice. Click the slice name
again to make the slice reappear on the chart.

.. image:: quality_alerts_report/pie-chart.png
   :align: center
   :alt: The pie chart graph view on the quality alerts report dashboard.

.. _quality_alerts_report/pivot-view:

Pivot view
----------

To display quality alert data in pivot view, click the :icon:`oi-view-pivot` :guilabel:`(Pivot)`
button in the top-right corner of the dashboard. The pivot view displays data in the form of a
spreadsheet.

> By default, each column is labeled with the name of a quality alert stage (ex. In Review). Since

Reviewer 2: ```suggestion
By default, each column is labeled with the name of a quality alert stage (e.g *In Review*). Since
```

alert stages are customizable, they may differ from one database to another.

By default, each row is labeled with the name of a quality alert team.

The cell at the intersection of a column and row displays a number, which signifies the number of

> quality alerts in the column's stage which are assigned to the row's quality alert team.

Reviewer 2: ```suggestion
quality alerts in the column's stage, which are assigned to the row's quality alert team.
```

To change the data displayed by the columns or rows, click the :guilabel:`Total` heading above
either. Then, click it again to open a popover menu, from which a new variable can be selected.

.. example::
   Click the :guilabel:`Total` heading above the columns to remove the stage names. Then, click the
   heading again, and select :guilabel:`Root Cause` from the popover menu. Each column is now titled
   with the name of a root cause.

.. figure:: quality_alerts_report/pivot-view.png
   :align: center
   :alt: The default pivot view on the quality alerts report dashboard.

   The default pivot view on the quality alerts report dashboard.

.. _quality_alerts_report/kanban-view:

Kanban view
-----------

To display quality alert data in a Kanban view, click the :icon:`oi-view-kanban`
:guilabel:`(Kanban)` button in the top-right corner of the dashboard.

> Kanban view displays all existing quality alerts in a Kanban board, with each alert organized into a

Reviewer 2: ```suggestion
The Kanban view displays all existing quality alerts in a Kanban board, with each alert organized into a
```

> specific stage (ex. In Review). Since kanban stages are customizable, they may differ from one

Reviewer 2: ```suggestion
specific stage (e.g. *In Review*). Since Kanban stages are customizable, they may differ from one
```

database to another.

> Quality alerts can be drag-and-dropped from one stage to another. Doing so changes the status of the

Reviewer 2: ```suggestion
Quality alerts can be dragged-and-dropped from one stage to another. Doing so changes the status of the
```

quality alert.

> Click on a quality alert to open it and see its details.

Reviewer 2: ```suggestion
Click on a quality alert to open it, and see its details.
```

.. image:: quality_alerts_report/kanban-view.png
   :align: center
   :alt: The Kanban view on the quality alerts report dashboard.

.. _quality_alerts_report/list-view:

List view
---------

To display all quality alerts in a list view, click the :icon:`oi-view-list` :guilabel:`(List)`
button in the top-right corner of the dashboard.

> List view displays all existing quality alerts in a list. Click on a quality alert to open it and

Reviewer 2: ```suggestion
The list view displays all existing quality alerts in a list. Click on a quality alert to open it, and
```

see its details.

.. image:: quality_alerts_report/list-view.png
   :align: center
   :alt: The quality alerts report list view.

.. _quality_alerts_report/calendar-view:

Calendar view
-------------

To display all quality alerts in a calendar view, click the :icon:`oi-view-calendar`
:guilabel:`(Calendar)` button in the top-right corner of the dashboard.

> By default, calendar view shows a calendar of the current month, with quality alerts listed on the

Reviewer 2: ```suggestion
By default, the calendar view shows a calendar of the current month, with quality alerts listed on the
```

day they are assigned. To switch to a different month, click the :icon:`oi-arrow-left`
:guilabel:`(left arrow)` or :icon:`oi-arrow-right` :guilabel:`(right arrow)` buttons at the top-left
corner of the dashboard.

To choose a different timescale, click the :guilabel:`Month` button to the right of the
:icon:`oi-arrow-right` :guilabel:`(right arrow)` button, then select :guilabel:`Day`,
:guilabel:`Week`, or :guilabel:`Year`, from the drop-down menu.

> On the calendar, click on a quality alert to open a popup displaying its details. Click the

Reviewer 2: ```suggestion
On the calendar, click on a quality alert to open a popover window displaying its details. Click the
```

> :guilabel:`Edit` button at the bottom of the popover to open the alert in a new page, or the

Reviewer 2: ```suggestion
:guilabel:`Edit` button at the bottom of the popover window to open the alert in a new page, or the
```
